theme/management/commands/migrate_quota_holder_from_irods.py

```diff
@@ -0,0 +1,59 @@
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+from django_irods.storage import IrodsStorage, SessionException
+from hs_core.models import BaseResource
+
+
+def get_quota_holder(res):
+    """Get quota holder of the resource.
+    return User instance of the quota holder for the resource or None if it does not exist
+    """
+    try:
+        uname = res.getAVU("quotaUserName")
+    except SessionException:
+        # quotaUserName AVU does not exist, return None
+        return None
+
+    if uname:
+        return User.objects.filter(username=uname).first()
+    else:
+        # quotaUserName AVU does not exist, return None
+        return None
+
+
+class Command(BaseCommand):
+    help = "Migrate quota_holder info from irods AVU into Django UserQuota model."
+
+    def handle(self, *args, **options):
+        istorage = IrodsStorage()
+
+        resources = BaseResource.objects.all()
+        number_of_res = resources.count()
+        counter = 1
+        res_with_error = []
+
+        if not istorage.exists(settings.IRODS_BAGIT_PATH):
+            print(f"iRODS path {settings.IRODS_BAGIT_PATH} does not exist.")
+            raise CommandError(f"iRODS path {settings.IRODS_BAGIT_PATH} does not exist.")
+
+        print(f"Total number of resources: {number_of_res}")
+        for res in resources:
```

Comment:
Since we are iterating over all resources in the system, it may be better to use the iterator here:
`for res in resources.iterator():`

Reply:
thanks, updated.